Tasket02

tasket opened the pull request:
Get dom0 restore working for v3 and v4 archives. This changes old restore path behavior to deposit entire dom0-home archive into a subdir.
_handle_dom0() signature is changed to accept stream (uncompressed tar archive).
https://github.com/QubesOS/qubes-issues/issues/3467

tasket committed:
Fix AdminVm class name

tasket committed:
Fix dom0 handling

tasket committed:
Fix dom0 restore

tasket committed:
style issues

tasket committed:
unused variable

marmarek commented on qubesadmin/backup/core2.py:
```diff
@@ -348,7 +351,7 @@ def load(self):
         self.globals['default_kernel'] = tree.getroot().get("default_kernel")
 
-        vm_classes = ["AdminVM", "TemplateVm", "TemplateHVm",
+        vm_classes = ["AdminVm", "TemplateVm", "TemplateHVm",
```
Ough, we definitely need some tests for this code...

tasket replied:
FWIW, I did perform manual tests on my 3.2 archives. I looked at the qubes.xml and saw that the elements were being tagged "AdminVm" not "AdminVM" (IIRC 4.0 xml saves with the latter case). Change results in R3.2 dom0 being picked up instead of ignored, and resulting in-memory class is "AdminVM" same as with R4.0 handling.

Not sure what errors we would anticipate with a test... maybe dom0 missing from the selection list (which was the original problem, actually).

tasket replied:
Note in core2.py:
~~~
        elif element.get('qid') == '0':
            kwargs['dir_path'] = element.get('dir_path')
            vm.klass = "AdminVM"
~~~

marmarek replied:
> Not sure what errors we would anticipate with a test... maybe dom0 missing from the selection list (which was the original problem, actually).

Yes, exactly this. And since the restore no longer override original user home, we can even compare restored data.

> Note in core2.py:

Yes, but it only iterates over elements listed in `vm_classes`...

marmarek committed:
backup: skip dom0's properties while restoring core2 backup

Fixes QubesOS/qubes-issues#3467

marmarek committed:
backup: add support for openssl 1.1.0 options

The command for listing available algorithms have changed:
old: openssl list-message-digest-algorithms
new: openssl list -digest-algorithms

marmarek committed:
tests: dom0 backup restore, both v3 and v4

QubesOS/qubes-issues#3467

marmarek committed:
backup: don't crash when no 'qubes' group is present

This is specifically the case on Travis-CI. But since dom0 backup is
restored into separate directory now, instead of directly overriding
user home, this check is much less relevant now.

marmarek commented:
Pushed tests here.
